graphql.go

```diff
@@ -122,6 +142,29 @@ func (c *Client) buildQueryAndOptions(op operationType, v any, variables map[str
 
 // Request the common method that send graphql request
 func (c *Client) request(ctx context.Context, query string, variables map[string]any, options *constructOptionsOutput) ([]byte, []byte, *http.Response, io.Reader, Errors) {
+	attempts := 0
+	for {
+		attempts++
+		rawData, extensions, resp, respReader, err := c.doRequest(ctx, query, variables, options)
+		if err == nil {
+			return rawData, extensions, resp, respReader, nil
+		}
+		if c.retries == 0 {
+			return rawData, extensions, resp, respReader, err
+		}
+		if attempts >= c.retries {
+			return rawData, extensions, resp, respReader, Errors{newError(ErrGraphQLMaxRetryExceeded, fmt.Errorf("after: %v attempts: %w", attempts, err))}
```

Comment:
```suggestion
			return rawData, extensions, resp, respReader, err
```

Wrapping the error isn't a good idea because the original information of the error will be lost. Developers can't unwrap the GraphQL error struct that cause breaking changes.

Reply:
I want to distinguish retry errors from other errors, so I updated the code to add a custom error to the error chain.